libs/auth/src/artifact_authentication.ts

```diff
@@ -0,0 +1,265 @@
+import { Buffer } from 'buffer';
```

Comment:
No need to review this commit in detail. I've just taken the methods on the `ArtifactAuthenticator` class and made them functions, reordering to avoid "this function is used before it's defined" errors.

I haven't changed the bodies of the functions at all, minus accessing config values from a `config` param instead of `this.`